bidding_auction_services_protected_app_signals_egress.md

```diff
@@ -17,71 +17,102 @@ The wire representation of the `egressPayload` will be noised per feature type.
 
 Based on the information in this document, adtechs can write parsers to prepare the `egressPayload` and transform the values it contains into features for use in their model training systems. 
 
-
 # Feature types
 
 We support two kinds of feature types: primitives, which can contain a single feature value; and collections, which can contain multiple primitives.
 
+**Nullability**
+
+Every feature type can optionally be made `nullable`, in which case the value `null` can be represented. By default features are _not_ nullable. A null value for a type is indicated by adding a signaling bit, 0 (for null) or 1 (for non-null), to the least-significant bit of the serialized representation. Consequently, the wire representation for a feature with nullable = true is always 1 bit larger than with nullable = false. 
+
+The contents of collections can be nullable and are handled as above. Collections themselves can also be nullable; handled the same way: a null value's wire representation is 0s for all value bits, with the 0 signaling bit.
 
 ## Primitive types
 
 These types represent single values: booleans, unsigned integers, and signed integers
 
-
 ### `boolean-feature-type`
 
+This represents a single boolean value.
 
-This represents a single boolean value
+Expected value: `true` or `false`, or optionally `null`
 
-Expected value: `true` or `false`
+Parameters:
+
+*   `nullable`: boolean indicating whether the `boolean-feature-type` can represent a null state or not. Defaults to `false`.
 
 **Wire format**
 
+Non-nullable:
+
 0 (`false`) or 1 (`true`)
 
+Null-able:
 
-### `unsigned-integer-feature-type`
+01 (`false`) or 11 (`true`), or 00 (`null`)
 
+### `unsigned-integer-feature-type`
 
 This represents a single non-negative integer value.
 
 Parameters:
 
-
-
 *   `size`: unsigned integer indicating the number of bits based on the range of values.
 
-Expected value:  non-negative integer value in the range [0,2^<sup>size</sup>-1]
+      * Expected value:  non-negative integer value in the range [0,2^<sup>size</sup>-1]
 
-For example, if `size `= `3` `unsigned-integer-feature-type` can have a value : ` [0, 7] `and will occupy 3 bits on the wire.
```

Comment:
Here and at other places we talk about the size on the wire without taking nullability into account.
See if making this conditional on nullability would make it more clear.

Reply:
I may have misunderstood, but this set of updates is written as if nullability does not affect the value of the size param; that the size param represents the bits of data to be communicated independent of whether there is also a null bit or not. Is that wrong? I will have to make some minor edits if my understanding was incorrect.

Reply:
discussed offline, my original understanding was correct. Updated explainer language to increase clarity.